### ISSUE TITLE
ICD11 --> ORDO mappings

### ISSUE THREAD

**Issue by joeflack4** [monarch-initiative/icd11#10]
## Overview
Produce some mappings.

**Comment by joeflack4** [comment]
@twhetzel Nico asked for this as a subtask of https://github.com/monarch-initiative/icd11/issues/10 back in January I think. For now I put it as 'very high'; This should come up at our check-in w/ Nico when we look at the board.
- https://github.com/monarch-initiative/icd11/issues/10

I don't actually know what this might entail other than running a SSSOM command on `icd11foundation.owl`.

**Comment by joeflack4** [comment]
FYI, I saw this during `build-mondo-ingest`. IDK what goal threw these warnings, but I take it that the reason is that the namespace does not exist in our Orphanet component. IDK for sure but I would imagine that this is not a new thing.

Example: `ERROR:root:Skipping statements(subject=Orphanet:1560,predicate=oio:hasDbXref,object=None,value=ICD-11:1F70 ,datatype=None,language=None,); ValueError: ICD-11:1F70  is not a valid URI or CURIE`

<details><summary>Full log</summary>
<p>

```
ERROR:root:Skipping statements(subject=Orphanet:1560,predicate=oio:hasDbXref,object=None,value=ICD-11:1F70 ,datatype=None,language=None,); ValueError: ICD-11:1F70  is not a valid URI or CURIE
ERROR:root:Skipping statements(subject=Orphanet:2073,predicate=oio:hasDbXref,object=None,value=ICD-11:7A20.0 ,datatype=None,language=None,); ValueError: ICD-11:7A20.0  is not a valid URI or CURIE
ERROR:root:Skipping statements(subject=Orphanet:2380,predicate=oio:hasDbXref,object=None,value=ICD-11:FB82.1 ,datatype=None,language=None,); ValueError: ICD-11:FB82.1  is not a valid URI or CURIE
ERROR:root:Skipping statements(subject=Orphanet:2582,predicate=oio:hasDbXref,object=None,value=ICD-11:4A43.Y ,datatype=None,language=None,); ValueError: ICD-11:4A43.Y  is not a valid URI or CURIE
ERROR:root:Skipping statements(subject=Orphanet:2764,predicate=oio:hasDbXref,object=None,value=ICD-11:FB82.1 ,datatype=None,language=None,); ValueError: ICD-11:FB82.1  is not a valid URI or CURIE
ERROR:root:Skipping statements(subject=Orphanet:293202,predicate=oio:hasDbXref,object=None,value=ICD-11:2B5F.2 ,datatype=None,language=None,); ValueError: ICD-11:2B5F.2  is not a valid URI or CURIE
ERROR:root:Skipping statements(subject=Orphanet:3005,predicate=oio:hasDbXref,object=None,value=ICD-11:LD24.1Y ,datatype=None,language=None,); ValueError: ICD-11:LD24.1Y  is not a valid URI or CURIE
ERROR:root:Skipping statements(subject=Orphanet:33276,predicate=oio:hasDbXref,object=None,value=ICD-11:2B57.1 ,datatype=None,language=None,); ValueError: ICD-11:2B57.1  is not a valid URI or CURIE
ERROR:root:Skipping statements(subject=Orphanet:36236,predicate=oio:hasDbXref,object=None,value=ICD-11:EA50.2 ,datatype=None,language=None,); ValueError: ICD-11:EA50.2  is not a valid URI or CURIE
ERROR:root:Skipping statements(subject=Orphanet:47045,predicate=oio:hasDbXref,object=None,value=ICD-11:4A60.1 ,datatype=None,language=None,); ValueError: ICD-11:4A60.1  is not a valid URI or CURIE
ERROR:root:Skipping statements(subject=Orphanet:521,predicate=oio:hasDbXref,object=None,value=ICD-11:2B33.2 ,datatype=None,language=None,); ValueError: ICD-11:2B33.2  is not a valid URI or CURIE
ERROR:root:Skipping statements(subject=Orphanet:545,predicate=oio:hasDbXref,object=None,value=ICD-11:2A80 ,datatype=None,language=None,); ValueError: ICD-11:2A80  is not a valid URI or CURIE
ERROR:root:Skipping statements(subject=Orphanet:572,predicate=oio:hasDbXref,object=None,value=ICD-11:4A01.12 ,datatype=None,language=None,); ValueError: ICD-11:4A01.12  is not a valid URI or CURIE
ERROR:root:Skipping statements(subject=Orphanet:67,predicate=oio:hasDbXref,object=None,value=ICD-11:1A36.00 ,datatype=None,language=None,); ValueError: ICD-11:1A36.00  is not a valid URI or CURIE
ERROR:root:Skipping statements(subject=Orphanet:69665,predicate=oio:hasDbXref,object=None,value=ICD-11:JA65.0 ,datatype=None,language=None,); ValueError: ICD-11:JA65.0  is not a valid URI or CURIE
ERROR:root:Skipping statements(subject=Orphanet:97360,predicate=oio:hasDbXref,object=None,value=ICD-11:LD24.A ,datatype=None,language=None,); ValueError: ICD-11:LD24.A  is not a valid URI or CURIE
```

</p>
</details>

**Comment by matentzn** [comment]
> FYI, I saw this during build-mondo-ingest. IDK what goal threw these warnings, but I take it that the reason is that the namespace does not exist in our Orphanet component. IDK for sure but I would imagine that this is not a new thing.

Dont worry about that error - I added a SPARQL query to add basically just copy that specific xref to one that is legal:
https://github.com/monarch-initiative/mondo-ingest/blob/8bcce8eeba57982128fd384e0ac55cdc053bd297/src/sparql/fix_xref_prefixes.ru#L38

You can ignore that warning.

> ORDO mappings: I don't actually know what this might entail other than running a SSSOM command on icd11foundation.owl.

You need to do some research.

- [x] 1. Check if ORDO contains ICD11 mappings - there should be a lot of them. Look at the file (components/ordo.owl) and convince yourself that the mappings are exact.
- [x] 2. Write a SPARQL query that reads ICD11 xrefs and asserts SKOS exact matches (there should be some sparql queries that already do this)
- [x] 3. Compare the mappings to whatever you can find in https://www.orphadata.com/alignments/

Alternatively, you can email Julie T from Orphanet, and ask her:

"What is the easiest way to get _exact_ mappings between ORDO and ICD11 Foundation?"

**Comment by matentzn** [comment]
(Also - make sure that whatever you look it is really ICD 11 foundation, and not linearization)

**Comment by joeflack4** [comment]
I checked off (1 - 3) above:
1. They're there, but to get foundation terms requires more work: https://github.com/monarch-initiative/icd11/pull/11#discussion_r1542176048
2. SPARQL query alone won't suffice. Going to ignore that for now. Also, we have `ordo-replace-annotation-based-mappings.ru`, which seems like it's supposed to be inserting `skos` mapping annotations, but I don't see those actually appearing on `components/ordo.owl`; it appears the query is broken. Also, I didn't end up needing the query. I was able to query the nomenclature file directly.
3. Done. Here's a report: https://github.com/monarch-initiative/icd11/pull/11#discussion_r1540220081
